modules/services/imagery_offset.js

```diff
@@ -0,0 +1,165 @@
+import * as d3 from 'd3';
+import _ from 'lodash';
+import rbush from 'rbush';
+import { geoExtent } from '../geo/index';
+import { utilQsString } from '../util/index';
+
+var apibase = 'http://offsets.textual.ru/get?',
+    inflight = {},
+    offsetCache;
+
+export default {
+    init: function() {
+        inflight = {};
+        offsetCache = rbush();
+        window.offsetCache = offsetCache;
+    },
+
+    reset: function() {
+        _.forEach(inflight, function(req) {
+            req.abort();
+        });
+        inflight = {};
+        offsetCache = rbush();
+    },
+    getImageryID: function(url, imageryType) {
```

Comment:
JavaScript probably contains more (or better) files for handling URLs. I'm wondering if it makes sense to write this differently, particularly when parsing the parameters part of the URL